OpenRA.Game/Graphics/SpriteRenderer.cs

```diff
@@ -151,9 +152,18 @@ public void DrawSprite(Sprite s, in float3 a, in float3 b, in float3 c, in float
 			nv += 6;
 		}
 
-		public void DrawVertexBuffer(IVertexBuffer<Vertex> buffer, int start, int length, PrimitiveType type, Sheet sheet, BlendMode blendMode)
+		public void DrawVertexBuffer(IVertexBuffer<Vertex> buffer, int start, int length, PrimitiveType type, IEnumerable<Sheet> sheets, BlendMode blendMode)
 		{
-			shader.SetTexture("Texture0", sheet.GetTexture());
+			var i = 0;
+			foreach (var s in sheets)
+			{
+				if (i >= SheetCount)
+					throw new ArgumentException("SpriteRenderer only supports {0} simultaneous textures".F(SheetCount), nameof(sheets));
```

Comment:
Its not clear to me how to translate that here, as the example uses a static method.

Is it the fact that its static that matters, or that it is calling another method (which doesnt need to be static)?

This will dictate whether we can use an anonymous method here or have to define a separate static method on the class.